### PR TITLE
bindata: automate bindata.go generation

### PR DESCRIPTION
**Pull request by iaguis**
We remove `bindata/bindata.go` and add `bindata/generate-bindata.go`.

At compile time we run `go generate`, which will find the directives in `bindata/generate-bindata.go` to create `bindata/bindata.go` from the files in `templates/`, so there's no need for the developer to do anything special.

### CODE REVIEW

**Comment by iaguis** [comment]
We can do this here (as opposed to k8scloudconfig) because we don't get `bindata.go` vendored anywhere.

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 .PHONY: verify-glide-installation install-vendor update-vendor all install check
 
 all:
+	go generate github.com/giantswarm/aws-operator/bindata
```
Comment (xh3b4sd) [inline Makefile]: What is this doing? I thought having the go generate in `bindata/generate-bindata.go` would be good enough?

Comment (iaguis) [inline Makefile]: I thought that too but it doesn't seem to be working 😞

Comment (iaguis) [inline Makefile]: All the examples in https://blog.golang.org/generate call `go generate` explicitly.

Comment (iaguis) [inline Makefile]: If that's the case I guess we need a change in architect? cc @JosephSalisbury

Comment (iaguis) [inline Makefile]: If we change architect, `go generate $(glide novendor)` seems to work

Comment (xh3b4sd) [inline Makefile]: /cc @JosephSalisbury

**Comment by asymmetric** [comment]
Do we still need this?

**Comment by iaguis** [comment]
> Do we still need this or will we implement the hook here too?

If we merge this it will be easier and we don't need the hook because the bindata files will get generated at compilation time.

However, we need architect to call `go-generate` if we want CI to pass.

**Comment by asymmetric** [comment]
But does it mean we need to remember to compile before committing, and add the `bindata.go` to the commit?

**Comment by iaguis** [comment]
> But does it mean we need to remember to compile before committing, and add the bindata.go to the commit?

No, `bindata.go` won't be checked-in.

**Comment by iaguis** [comment]
We won't be using go-bindata.

**Pull request closed without merging**